acceptance-tests/tests/src/test/java/org/hyperledger/besu/tests/acceptance/plugins/PermissioningPluginTest.java

```diff
@@ -0,0 +1,103 @@
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.tests.acceptance.plugins;
+
+import org.hyperledger.besu.ethereum.core.Hash;
+import org.hyperledger.besu.tests.acceptance.dsl.AcceptanceTestBase;
+import org.hyperledger.besu.tests.acceptance.dsl.account.Account;
+import org.hyperledger.besu.tests.acceptance.dsl.blockchain.Amount;
+import org.hyperledger.besu.tests.acceptance.dsl.node.BesuNode;
+import org.hyperledger.besu.tests.acceptance.dsl.node.configuration.BesuNodeConfigurationBuilder;
+import org.hyperledger.besu.tests.acceptance.dsl.transaction.account.TransferTransaction;
+
+import java.util.Collections;
+
+import org.junit.Before;
+import org.junit.Test;
+
+public class PermissioningPluginTest extends AcceptanceTestBase {
+  private BesuNode minerNode;
+
+  private BesuNode aliceNode;
+  private BesuNode bobNode;
+  private BesuNode charlieNode;
+
+  @Before
+  public void setUp() throws Exception {
+    BesuNodeConfigurationBuilder builder =
+        new BesuNodeConfigurationBuilder()
+            .plugins(Collections.singletonList("testPlugins"))
+            .jsonRpcEnabled()
+            .jsonRpcTxPool()
+            .jsonRpcAdmin();
+
+    minerNode = besu.create(builder.name("miner").miningEnabled().build());
+
+    aliceNode = besu.create(builder.name("alice").miningEnabled(false).keyFilePath("key").build());
+
+    bobNode = besu.create(builder.name("bob").miningEnabled(false).keyFilePath("key1").build());
+
+    charlieNode =
+        besu.create(builder.name("charlie").miningEnabled(false).keyFilePath("key2").build());
+
+    cluster.start(minerNode, charlieNode);
+
+    cluster.startNode(aliceNode);
+    aliceNode.verify(net.awaitPeerCount(2));
+    cluster.startNode(bobNode);
+    bobNode.verify(net.awaitPeerCount(2));
+  }
+
+  @Test
+  public void blockedConnectionNodeCanOnlyConnectToTransactionNode() {
+    minerNode.verify(admin.hasPeer(aliceNode));
+    minerNode.verify(admin.hasPeer(bobNode));
+    minerNode.verify(admin.hasPeer(charlieNode));
+
+    aliceNode.verify(admin.hasPeer(minerNode));
+    aliceNode.verify(admin.doesNotHavePeer(bobNode));
+    aliceNode.verify(admin.hasPeer(charlieNode));
+
+    bobNode.verify(admin.hasPeer(minerNode));
+    bobNode.verify(admin.doesNotHavePeer(aliceNode));
+    bobNode.verify(admin.hasPeer(charlieNode));
+
+    charlieNode.verify(admin.hasPeer(minerNode));
+    charlieNode.verify(admin.hasPeer(aliceNode));
+    charlieNode.verify(admin.hasPeer(bobNode));
+  }
+
+  @Test
+  public void transactionsAreNotSendToBlockPendingTransactionsNode() {
+    final Account account = accounts.createAccount("account-one");
+    final Amount balance = Amount.ether(20);
+
+    TransferTransaction tx = accountTransactions.createTransfer(account, balance);
+
+    Hash txHash = aliceNode.execute(tx);
+
+    minerNode.execute(minerTransactions.minerStop());
```

Comment:
does this test rely on the mining stopping before the tx we just sent is mined? Won't that be flaky?

Reply:
Good spot, thanks